esphome/components/network/__init__.py

```diff
@@ -80,6 +118,53 @@ async def to_code(config):
     cg.add_define("USE_NETWORK")
     if CORE.using_arduino and CORE.is_esp32:
         cg.add_library("Networking", None)
+
+    # Apply high performance networking settings if requested by any component
+    if (
+        CORE.is_esp32
+        and CORE.using_esp_idf
+        and CORE.data.get(KEY_HIGH_PERFORMANCE_NETWORKING, False)
+    ):
+        # Check if PSRAM is guaranteed (set by psram component during final validation)
+        psram_guaranteed = CORE.data.get(KEY_ESP32, {}).get(KEY_PSRAM_GUARANTEED, False)
+
+        if psram_guaranteed:
```

Comment:
nit: debug logging would be nice here..

```suggestion
        if psram_guaranteed:
            _LOGGER.debug("High performance networking: aggressive settings (PSRAM guaranteed)")
        
```

Reply:
Maybe even log at info to make it obvious considering how much RAM it can use